PreFreeSurfer/PreFreeSurferPipeline.sh

```diff
@@ -244,6 +244,12 @@ for TXw in ${Modalities} ; do
     ${RUN} ${PipelineScripts}/AnatomicalAverage.sh -o ${TXwFolder}/${TXwImage} -s ${TXwTemplate} -m ${TemplateMask} \
     -n -w ${TXwFolder}/Average${TXw}Images --noclean -v -b $BrainSize $OutputTXwImageSTRING
     #fi
+   ###Added by Bene to use created warp above and apply it to the brain 
+    ${FSLDIR}/bin/fslmaths ${TXwFolder}/${TXwImage}_brain -bin ${TXwFolder}/${TXwImage}_mask
+    ${FSLDIR}/bin/fslmaths ${TXwFolder}/${TXwImage}_brain -bin ${TXwFolder}/${TXwImage}_mask_rot2first  #take this out once done testing just something for QC for now
+    $FSLDIR/bin/applywarp --rel -i ${TXwFolder}/${TXwImage}_mask --premat=${TXwFolder}/Average${TXw}Images/ToHalfTrans0001.mat -r ${TXwFolder}/${TXwImage} -o ${TXwFolder}/${TXwImage}_mask --interp=nn
+    ${FSLDIR}/bin/fslmaths ${TXwFolder}/${TXwImage} -mas ${TXwFolder}/${TXwImage}_mask ${TXwFolder}/${TXwImage}_brain
+    ### Done with edits by Bene
```

Comment:
Here I am taking the brain that was averaged in the preliminary masking stage, and binarizing it to make a mask. I then apply the warp created in the AnatomicalAverage.sh code that brings the first T1 to the halfway point to this mask, as the T1w_average that made this brain in the preliminary masking stage used the first t1 as a reference to average. Than I take the newly registered to halfway point mask, and mask the T1w average made in the AnatomicalAverage stage. (Correction: T1 or T2 as it loops through these TX)

Reply:
Bene -- I'm not finding this file: ToHalfTrans0001 anywhere else in this file. Are you sure it will exist at this point?
The output from AnatomicalAverage.sh is called $OutputTXwImageSTRING and I'm not sure what that resolves to at this point, but it's going to be something like ${TXwFolder}/${TXwImage} followed by some number and then "_gdc". So, I don't know where ToHalfTrans0001.mat is coming from.
Be sure to test with an empty directory - you might have that file left from one of your tests.

Reply:
@b3n3dikt I concur with @kathy-snider that I don't see the ToHalfTrans0001 file either in this snippet and it looks like an expected input.